homeassistant/components/goodwe/sensor.py

```diff
@@ -175,16 +202,21 @@ def __init__(
         self.entity_description = _DESCRIPTIONS.get(sensor.unit, DIAG_SENSOR)
         if not self.entity_description.native_unit_of_measurement:
             self._attr_native_unit_of_measurement = sensor.unit
+        if self.entity_description == DIAG_SENSOR and (
+            "Enum" in type(sensor).__name__ or sensor.id_ == "timestamp"
+        ):
+            self.entity_description = ENUM_SENSOR
```

Comment:
Make a function that returns the correct entity description for this sensor instead. It's weird to first set the description to something and then have a condition to change it.

Reply:
Done, the entity description is no longer re-assigned, the description and unit lookup was made more straightforward.